cmd/proximityBlockVcf/proximityBlockVcf.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/vcf"
+	"log"
+	"math/rand"
+)
+
+func proximityBlockVcf(inFile string, outFile string, distance int, randSeed bool, setSeed int64) {
+	var err error
+	common.RngSeed(randSeed, setSeed)
+	records, header := vcf.Read(inFile)
+	rand.Shuffle(len(records), func(i, j int) { records[i], records[j] = records[j], records[i] })
+
+	//make an output slice and put the first entry in it.
+	var retainedVcfs []vcf.Vcf = make([]vcf.Vcf, 1)
+	retainedVcfs[0] = records[0]
+
+	for i := 1; i < len(records); i++ {
+		if vcfPassesDistanceThreshold(retainedVcfs, records[i], distance) {
+			retainedVcfs = append(retainedVcfs, records[i])
+		}
+	}
+
+	out := fileio.EasyCreate(outFile)
+	vcf.NewWriteHeader(out, header)
+	for _, j := range retainedVcfs {
+		vcf.WriteVcf(out, j)
+	}
+	err = out.Close()
+	exception.PanicOnErr(err)
+}
+
+func vcfPassesDistanceThreshold(retainedVcfs []vcf.Vcf, i vcf.Vcf, distanceThreshold int) bool {
+	for _, j := range retainedVcfs {
+		if i.Chr == j.Chr {
+			if i.Pos < j.Pos {
+				if j.Pos-i.Pos < distanceThreshold {
+					return false
+				}
+			} else {
+				if i.Pos-j.Pos < distanceThreshold {
+					return false
+				}
+			}
+		}
+	}
+
+	return true
+}
+
+func usage() {
+	fmt.Print(
+		"proximityBlockVcf - Pseudorandomly selects variants" +
+			"from an input VCF file and retains variants in the output" +
+			"that do not fall within a user-specified distance" +
+			"of variants already chosen. Output is returned in a shuffled order.\n" +
+			"Usage:\n" +
+			"proximityBlockVcf input.vcf output.vcf distance" +
+			"options:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	var randSeed *bool = flag.Bool("randSeed", false, "Uses a random seed for the RNG.")
```

Comment:
I ended up needing to google around to figure out that RNG stands for 'random number generator'. Not a big deal, probably most know this acronym, but I think generally it'd benefit new users to type out acronyms within usage statements.